sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import scala.collection.immutable.Seq
+
+import org.apache.spark.sql.internal.SqlApiConf
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.StringType
+
+// scalastyle:off nonascii
+class CollationSQLExpressionsSuite
+  extends QueryTest
+    with SharedSparkSession {
+
+  test("Conv expression with collation") {
+    // Supported collations
+    case class ConvTestCase(
+        num: String,
+        from_base: String,
+        to_base: String,
+        collationName: String,
+        result: String)
+
+    val testCases = Seq(
+      ConvTestCase("100", "2", "10", "UTF8_BINARY", "4"),
+      ConvTestCase("100", "2", "10", "UTF8_BINARY_LCASE", "4"),
+      ConvTestCase("100", "2", "10", "UNICODE", "4"),
+      ConvTestCase("100", "2", "10", "UNICODE_CI", "4")
+    )
+    testCases.foreach(t => {
+      val query =
+        s"""
+           |select conv(collate('${t.num}', '${t.collationName}'), ${t.from_base}, ${t.to_base})
```

Comment:
why do we add end-to-end tests but not unit tests?

Reply:
I think these expressions are only pass-through (changes only in `inputTypes`/`dataType`) in mathExpressions.scala, so these tests should be enough